### ISSUE TITLE
Discovery: include word proximity in search results

### ISSUE THREAD

**Issue by pdpinch** [mitodl/lore#869]
in multi-word searches, is there a way we can weight search results higher when the words appear in close proximity? 

Another approach to #866

**Comment by pdpinch** [comment]
@pwilkins does this make sense?

**Comment by pwilkins** [comment]
Yes, Elasticsearch supports proximity.  We "simply" have to add proximity support to the tool chain that generates an index.  (The tool chain is called an analyzer.)  This article section provides a high-level description of how this feature works. 

https://www.elastic.co/blog/found-text-analysis-part-1#optimizing-phrase-searches-with-shingles

As @giocalitri mentioned in our f2f meeting, we should be prepared for this to create larger index files.  Perhaps significantly larger.
